### DISCUSSION TITLE
Go Relayer Issue

### DISCUSSION THREAD

**Discussion by torvald2** [cosmos/community#88] [IBC]
I have issue with IBC and Go Relayer 

When I trying send IBC Package got error "module does not own channel capability” in my observer module. 
Capabilities have to store on path capabilities/ports/transfer/channels/channel-0
But when I created connection there no OnChanOpenInit call in my module 

My Go replayer steps 
rly chains add -f ./test_config/subnode_go_relayer.json
rly chains add -f ./test_config/government_go_ralayer.json
rly keys add government_go_ralayer test
rly keys add subnode_go_relayer test   
rly paths new subnode government observer-path
rly transact clients observer-path 
rly transact connection observer-path     
  rly transact channel observer-path   
  rly start observer-path    

It’s creates channel-0 and port id transfer 

Command subnetwork-noded  query ibc channel channels --chain-id subnode --home ~/.subnode 
Shows 

channels:
- channel_id: channel-0
  connection_hops:
  - connection-0
  counterparty:
    channel_id: channel-0
    port_id: transfer
  ordering: ORDER_UNORDERED
  port_id: transfer
  state: STATE_OPEN
  upgrade_sequence: "0"
  version: ics20-1
height:
  revision_height: "3746"
  revision_number: “0"

Channel and port is exists but there no capability for my module 
It’s makes me crazy Help please ****

**Comment by reecepbcups** [comment]
@torvald2 Hey! Was just forwarded this message from someone else

When you are building an IBC module, you need to claim the capability in your keeper. An example is like so

```
func (k Keeper) BindPort(ctx sdk.Context, portID string) error {
	capability := k.portKeeper.BindPort(ctx, portID)
	return k.ClaimCapability(ctx, capability, host.PortPath(portID))
}
```

This uses the `github.com/cosmos/ibc-go/modules/capability` module for v50, or cosmos-sdk/x/capability in < v47

A good example is in the [ibc-go repo with the transfer keeper](https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/keeper/keeper.go#L103-L126)
